proto/penumbra/penumbra/core/transaction/v1alpha1/transaction.proto

```diff
@@ -20,7 +20,7 @@ message Transaction {
 
 // A transaction ID, the Sha256 hash of a transaction.
 message Id {
-  bytes hash = 1;
+  string hash = 1;
```

Comment:
The transaction hash should be `bytes`, so that we transmit it as binary data. Otherwise, the proto doesn't specify the data format properly, because it doesn't specify how the string encodes the hash (is it base64? hex? something else?)

We should leave the encoding as `bytes`, and fix any translation errors where they occur, rather than propagating the transation errors into our own data modeling.

Reply:
Heard, and accounted for in a later debugging branch (#2852) which leaves the proto hash field as-is.

`pcli` is currently operating under the assumption that the bytes field in this request proto will be populated from a string, and is relying on .parse()/from_str to accordingly and successfully interpret those bytes.

In #2852 as a debugging sanity check I was able to successfully read the `hash` field in the proto (using the web gRPC UI) as copied from pcli output, but doing so breaks `pcli v tx` insofar as it starts failing to return transaction info from the client.

I'm out of time today, but for tomorrow:

1. Implement your suggestion (https://github.com/penumbra-zone/penumbra/issues/2844#issuecomment-1641860008) of adding an alt_hex field to the proto & ensure that the client code continues to work with pcli while also supporting retrieving transaction info from hex-encoded strings.
2. Once both sides of this (pcli and pclientd-via-webgrpcui) are working reliably, annotate the various field encodings appropriately so that we know which data format should be specified/what people should be passing to each optional field.